libs/vault/src/components/assign-collections.component.ts

```diff
@@ -129,10 +128,18 @@ export class AssignCollectionsComponent implements OnInit {
     );
 
   protected transferWarningText = (orgName: string, itemsCount: number) => {
-    const pluralizedItems = this.pluralizePipe.transform(itemsCount, "item", "items");
-    return orgName
-      ? this.i18nService.t("personalItemsWithOrgTransferWarning", pluralizedItems, orgName)
-      : this.i18nService.t("personalItemsTransferWarning", pluralizedItems);
+    const haveOrgName = !!orgName;
+
+    switch (true) {
+      case itemsCount > 1 && haveOrgName:
+        return this.i18nService.t("personalItemsWithOrgTransferWarningPlural", itemsCount, orgName);
+      case itemsCount > 1 && !haveOrgName:
+        return this.i18nService.t("personalItemsTransferWarningPlural", itemsCount);
+      case itemsCount === 1 && haveOrgName:
+        return this.i18nService.t("personalItemWithOrgTransferWarningSingular", orgName);
+      default:
+        return this.i18nService.t("personalItemTransferWarningSingular");
+    }
```

Comment:
Is there a preference to write `switch` statements? I feel like it reads a bit more complex than just a plain `if` with  early `returns`.  Maybe not though? There is less noise from braces.

```suggestion
    if (itemsCount > 1 && haveOrgName) {
      return this.i18nService.t("personalItemsWithOrgTransferWarningPlural", itemsCount, orgName);
    }

    if (itemsCount > 1 && !haveOrgName) {
      return this.i18nService.t("personalItemsTransferWarningPlural", itemsCount);
    }

    if (itemsCount === 1 && haveOrgName) {
      return this.i18nService.t("personalItemWithOrgTransferWarningSingular", orgName);
    }

    return this.i18nService.t("personalItemTransferWarningSingular");
```

Reply:
I'll leave it to a bitwarden eng for the common pattern and which to go with. My mental flag is usually 3 `if`s changes to a switch.

Reply:
We generally don't have a preference, but in this case, I would favour the if with early returns cause the conditions are few, and I doubt any new condition would be added in future

Reply:
Updated! [232e043](https://github.com/bitwarden/clients/pull/10205/commits/232e043f1fa20294acf6340913944c8bad544767)